docs/architecture/commitlog/fileformat.rst

```diff
@@ -0,0 +1,63 @@
+ScyllaDB commitlog segment file format
+======================================
+
+**Note:** Commitlog file formats are subject to change between scylla versions. Users should not make assumptions about nor rely on them.
+Commitlog files should *never* be used across ScyllaDB updates. This information is provided mainly for ScyllaDB contributors.
+
+File descriptor structure
+.........................
+
+ScyllaDB commitlog segment files are named with a versioned, time-indexed scheme, as::
+
+        <Prefix><version>-<id>.log
+
+Where `<Prefix>` is application specific, but typically "Commitlog-", or in case of 
+files being recycled "Recycled-Commitlog-", `<version>` is the file format version,
+and `<id>` is the id part of a replay position (timestamp + shard).
```

Comment:
At this point, the reader really has no idea what is a "replay position" is, or what parts it has besides "id" :-(

Reply:
Again, I am just documenting the file format here, by request. This is not intended to be (yet) a through explanation of all aspects of neither CL not mutation application nor truncation handling.